integrate modal into appointmentList

pmar0 opened the pull request:
- added modalAddView, a view that allows routing to have modal over dash (easiest way to trigger modal, also allows for better link sending)
- added placeholder "loading..." for appointmentList when loading and also check to see if there's no appointments, which displays a different message
- tweaked card to ensure it works with appointmentList
- set up modal error states and error onChange clearing
- implemented proper error catching from server + display to show validations (this presented a problem, will list below)
- eliminated much of the modal boolean trigger functionality, as we are using routing instead
- installed cookie-parser on server end and imported into server.ts in order to get JWT auth working
- changed export for JWT auth so it would work properly
- changed app controller bad responses, were code 500 and now are 400
- added auth requirement to all appointment paths and made sure that all current axios requests have "{withCredentials: true}" when needed. This is fully functional from my testing of display (get all) and create (post)
- added an appointmentPost interface to fix TS error that was complaining that there is no _id, as that was added in the previous PR

Major issue is  validations on the modal result in it going all skinny n weird due to block styling of validation errors making the modal too tall, maybe? Not sure, poked at it for all too long but Ima have to leave that to y'all. I can swing back to integrate edit and delete pretty quickly, but I just want that modal working first. @kishanpatel97 you might have an easier time figuring it out since you created the modal. **I currently have the alerts in as spans so things dont break immediately. Just change the validation spans back to p tags and you'll see what I mean.**

Resolves #44 not going to consider it solving the connection of card to modal yet, although that's an easy step.

Edit, added more because of everything that went on, see below:
commit d6f565f

- fixed modal validation issue 
- added edit and delete functionality
- set up edit functionality to default to add in case of error

Resolves #42 #19

pmar0 committed:
integrate modal into appointmentList

get JWT auth functional

b-chang commented on client/src/components/appointmentList.tsx:
```diff
@@ -12,8 +11,9 @@ const AppointmentList: React.FC = (props) => {
-      const response =  await axios.get<IAppointment[]>('http://localhost:8000/api/appointments')
+      const response =  await axios.get<IAppointment[]>('http://localhost:8000/api/appointments',{withCredentials:true})
       setAppointments(prev => [...prev, ...response.data])
+      console.log(response.data)
```
should probably remove this

b-chang replied:
or did i leave this in?

pmar0 replied:
I added when testing, forgot to remove last night..removed.

b-chang commented on client/src/components/modal.tsx:
```diff
@@ -1,16 +1,23 @@
 const Modal: React.FC = () => {
-    const [showModal, setShowModal] = useState(false);
-    const [errors, setErrors] = useState({});
+	const navigate = useNavigate();
```
tabbing

pmar0 replied:
navigate actually somehow had a tab less..VSCode being weird

b-chang commented on client/src/components/modal.tsx:
```diff
@@ -40,122 +51,126 @@ const Modal: React.FC = () => {
                     notes: '',
                 });
-                setErrors('');
+                setErrors({
```
is this the actually tabbing? or is this just github being weird

pmar0 replied:
think I fixed...more VSCode tabbing fun!

pmar0 committed:
brandon told me to fix stuff :P

Commit:
fixed formatting woes mentioned by @b-chang

pmar0 committed:
fixed modal validation issue
added edit and delete functionality
set up edit functionality to default to add in case of error

pmar0 committed:
Merge branch 'connectModal' of https://github.com/kishanpatel97/APPT into connectModal

kishanpatel97 reviewed (approved):
Looks hawt

Commit:
minor formatting changes

kishanpatel97 reviewed (approved):
looks good. reformatted
